planner/core/task.go

```diff
@@ -968,7 +985,14 @@ func (p *PhysicalStreamAgg) attach2Task(tasks ...task) task {
 	t := tasks[0].copy()
 	inputRows := t.count()
 	if cop, ok := t.(*copTask); ok {
-		partialAgg, finalAgg := p.newPartialAggregate()
+		// copToFlash means whether the cop task is run on flash storage
+		copToFlash := false
+		if cop.tablePlan != nil {
+			if ts, ok := cop.tablePlan.(*PhysicalTableScan); ok {
```

Comment:
Is it possible the the child is `selection`?

Reply:
Oh. It's possible.

```sql
tidb(localhost:4000) > desc select count(b) from t1 where t1.b=1 group by a;
+--------------------------+----------+-----------+-------------------------------------------------------------+
| id                       | count    | task      | operator info                                               |
+--------------------------+----------+-----------+-------------------------------------------------------------+
| HashAgg_13               | 8.00     | root      | group by:Column#7, funcs:count(Column#6)                    |
| └─TableReader_14         | 8.00     | root      | data:HashAgg_6                                              |
|   └─HashAgg_6            | 8.00     | cop[tikv] | group by:Column#1, funcs:count(Column#2)                    |
|     └─Selection_12       | 10.00    | cop[tikv] | eq(Column#2, 1)                                             |
|       └─TableScan_11     | 10000.00 | cop[tikv] | table:t1, range:[-inf,+inf], keep order:false, stats:pseudo |
+--------------------------+----------+-----------+-------------------------------------------------------------+
5 rows in set (0.00 sec)
```